tests/fixture/testnet/config.go

```diff
@@ -0,0 +1,353 @@
+// AvalancheGo node.
+type NodeConfig struct {
+	NodeID ids.NodeID
+	Flags  FlagsMap
+}
+
+func NewNodeConfig() *NodeConfig {
+	return &NodeConfig{
+		Flags: FlagsMap{},
+	}
+}
+
+// Convenience method for setting networking flags.
+func (nc *NodeConfig) SetNetworkingConfig(
+	httpPort int,
+	stakingPort int,
+	bootstrapIDs []string,
+	bootstrapIPs []string,
+) {
+	startDefaults := FlagsMap{
+		cfg.HTTPPortKey:    httpPort,
+		cfg.StakingPortKey: stakingPort,
+	}
+	if len(bootstrapIDs) > 0 {
+		startDefaults[cfg.BootstrapIDsKey] = strings.Join(bootstrapIDs, ",")
+		startDefaults[cfg.BootstrapIPsKey] = strings.Join(bootstrapIPs, ",")
+	}
+	nc.Flags.SetDefaults(startDefaults)
+}
+
+// Ensures staking and signing keys are generated if not already present.
+func (nc *NodeConfig) EnsureKeys() error {
+	err := nc.EnsureBLSSigningKey()
+	if err != nil {
+		return err
+	}
+	return nc.EnsureStakingKeypair()
+}
+
+// Ensures a BLS signing key is generated if not already present.
+func (nc *NodeConfig) EnsureBLSSigningKey() error {
+	// Attempt to retrieve an existing key
+	existingKey, err := nc.Flags.GetStringVal(cfg.StakingSignerKeyContentKey)
+	if err != nil {
+		return err
+	}
+	if len(existingKey) > 0 {
+		// Nothing to do
+		return nil
+	}
+
+	// Generate a new signing key
+	newKey, err := bls.NewSecretKey()
+	if err != nil {
+		return fmt.Errorf("failed to generate staking signer key: %w", err)
+	}
+	nc.Flags[cfg.StakingSignerKeyContentKey] = base64.StdEncoding.EncodeToString(newKey.Serialize())
+	return nil
+}
+
+// Ensures a staking keypair is generated if not already present and that the node ID is set.
+func (nc *NodeConfig) EnsureStakingKeypair() error {
+	keyKey := cfg.StakingTLSKeyContentKey
+	certKey := cfg.StakingCertContentKey
+
+	key, err := nc.Flags.GetStringVal(keyKey)
+	if err != nil {
+		return err
+	}
+
+	cert, err := nc.Flags.GetStringVal(certKey)
+	if err != nil {
+		return err
+	}
+
+	if len(key) == 0 && len(cert) == 0 {
+		// Generate new keypair
+		tlsCertBytes, tlsKeyBytes, err := staking.NewCertAndKeyBytes()
+		if err != nil {
+			return fmt.Errorf("failed to generate staking keypair: %w", err)
+		}
+		nc.Flags[keyKey] = base64.StdEncoding.EncodeToString(tlsKeyBytes)
+		nc.Flags[certKey] = base64.StdEncoding.EncodeToString(tlsCertBytes)
+	} else if !(len(key) > 0 && len(cert) > 0) {
+		// Only one of key and cert was provided
+		return fmt.Errorf("%q and %q must be provided together or not at all", keyKey, certKey)
+	}
+
+	err = nc.EnsureNodeID()
+	if err != nil {
+		return fmt.Errorf("failed to derive a node ID: %w", err)
+	}
+
+	return nil
+}
+
+// Attempt to derive the node ID from the node configuration.
+func (nc *NodeConfig) EnsureNodeID() error {
+	keyKey := cfg.StakingTLSKeyContentKey
+	certKey := cfg.StakingCertContentKey
+
+	key, err := nc.Flags.GetStringVal(keyKey)
+	if err != nil {
+		return err
+	}
+	if len(key) == 0 {
+		return fmt.Errorf("missing value for %q", keyKey)
```

Comment:
shouldn't we check for this error sooner? can we just add a Validate method to Flags for required keys? Otherwise can we make this error more clear as to the root cause?